### ISSUE TITLE
Current time and buffered position are greater than duration(HLS) 

### ISSUE THREAD

**Issue by HSY229** [google/ExoPlayer#1930]
1) Add the HLS sample into `media.exolist.json`:

```
"name": "HLS",
    "samples": [
      {
        "name": "wrong EXTINF sample",
        "uri": "http://laria.space/WANG_DA_BIN/WANG_DA_BIN_02.m3u8"
      },
```

2) Launch ExoPlayer demo app
3) press "HLS" button
4) press "wrong EXTINF sample" button

When video is playback in the first segment,
we can see the secondary progress is fill the progress bar;

```
package com.google.android.exoplayer2.ui;

public class PlaybackControlView extends FrameLayout {
    ......
    private void updateProgress() {
            ......
            long bufferedPosition = player == null ? 0 : player.getBufferedPosition();
            // bufferedPosition is 2649230, duration is 1650000
            progressBar.setSecondaryProgress(progressBarValue(bufferedPosition));
            ......
    }
    ......
}
```

Then, when it ready to play the second segment,
the current time text is greater than duration,
and seek bar jump to the end by itself. 

I think the problem is come from this m3u8,
its `#EXTINF` tag mark every segment to be 10 seconds.

I use Safari and VLC-Android to play the same video,
but can not reproduce the problem.

ExoPlayer demo project r2.0.2
Android Studio 2.2
gradle-2.14.1-all.zip
com.android.tools.build:grade:2.2.0
buildToolsVersion='24.0.3'
compileSdkVersion=24
targetSdkVersion=24

Samsung Galaxy S5 6.0.1 
Samsung Galaxy S6 6.0.0 (This is provide by Genymotion)
RedMi Android 4.4.2

[log.txt](https://github.com/google/ExoPlayer/files/523435/log.txt)

![secondary](https://cloud.githubusercontent.com/assets/5857510/19295593/359e931c-9068-11e6-886d-22c8b2aa2ce1.png)
![currenttime](https://cloud.githubusercontent.com/assets/5857510/19295595/3798313c-9068-11e6-8173-28cdb3ec99e0.png)

**Comment by ojw28** [comment]
If the HLS playlists contains incorrect information, surely this is a content issue rather than a player issue. Please clarify.

**Comment by HSY229** [comment]
Thanks for your replay.

Actually this is the contain problem.
But if ExoPlayer can handle it, 
ExoPlayer will be better.

**Comment by AquilesCanta** [comment]
There is nothing wrong with the media playlists, I think. The issue is that PTS's from different variants do not match. When exoplayer adapts(in your case, it happens in the second segemnt), the discontinuity is found in the audio timestamps, making the player think that the position has changed drastically. Unfortunately, there is not much we can do about this. Having no issue with other players does not necessarily mean it will not occur, as it depends on adaptation.

I say we cannot do much, because this is against the spec, and supporting media that is not spec compliant is bad for everyone. In the last available version of the [HLS spec, section 6.2.4](https://tools.ietf.org/html/draft-pantos-http-live-streaming-20#section-6.2.4):

> Each Variant Stream MUST present the same content.
> 
> Matching content in Variant Streams MUST have matching timestamps.
> allows clients to synchronize the media.

**Comment by HSY229** [comment]
@AquilesCanta Thank you for the explanation and guidance.
